scripts/dropbox/migrate_to_external_accounts.py

```diff
@@ -0,0 +1,220 @@
+            ))
+        else:
+            external_accounts_created += 1
+            linked_node_settings_documents = database['dropboxnodesettings'].find({
+                'user_settings': user_settings_document['_id']
+            })
+            if not user or not user.is_active:
+                if linked_node_settings_documents.count():
+                    logger.warn("DropboxUserSettings<_id:{0}> has no owner, but is used by DropboxNodeSettings: {1}.".format(
+                        user_settings_document['_id'],
+                        ', '.join(linked_node_settings_documents.map(lambda d: d['_id']))
+                    ))
+                    raise RuntimeError("This should never happen.")
+                else:
+                    logger.info("DropboxUserSettings<_id:{0}> either has no owner or the owner's account is not active, and will not be migrated.".format(
+                        user_settings_document['_id']
+                    ))
+                    continue
+            else:
+                user_settings_instance = make_new_user_settings(user)
+                for node_settings_document in linked_node_settings_documents:
+                    verify_node_settings_document(node_settings_document)
+                    if node_settings_document['deleted']:
+                        logger.info(
+                            "Found dropboxnodesettings document (id:{0}) that is marked as deleted. It will be hard-deleted during the migration.".format(
+                                node_settings_document['_id']
+                            )
+                        )
+                        continue
+                    node = Node.load(node_settings_document['owner'])
+                    if not node:
+                        logger.info("DropboxNodeSettings<_id:{0}> either has no associated Node, and will not be migrated.".format(
+                            node_settings_document['_id']
+                        ))
+                        continue
+                    else:
+                        make_new_node_settings(
+                            node,
+                            node_settings_document,
+                            external_account,
+                            user_settings_instance
+                        )
+                        migrated_node_settings += 1
+        migrated_user_settings += 1
+    logger.info(
+        "Created {0} new external accounts, migrated {1} dropboxusersettings, and migrated {2} dropboxnodesettings.".format(
+            external_accounts_created, migrated_user_settings, migrated_node_settings
+        )
+    )
+    # Migrate dropbnodesettings without a usersettings document
+    unauthorized_node_settings_documents = database['dropboxnodesettings'].find({
+        'user_settings': 'null'
+    })
+    for node_settings_document in unauthorized_node_settings_documents:
+        make_new_node_settings(node, node_settings_document)
```

Comment:
This might not even be necessary; it's not being hit when migrating staging data. I think the conditions required to be here may represent an invalid database state. 
(`node` <--> `[addon]_node_settings` <-/no associated/-> `user_settings` <--> `user` 
implies `node` <-/no associated/-> `user`, which is invalid)